src/code.cloudfoundry.org/cni-wrapper-plugin/netrules/masquerader_test.go

```diff
@@ -0,0 +1,316 @@
+package netrules_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"code.cloudfoundry.org/cni-wrapper-plugin/fakes"
+	"code.cloudfoundry.org/cni-wrapper-plugin/lib"
+	"code.cloudfoundry.org/cni-wrapper-plugin/netrules"
+	lib_fakes "code.cloudfoundry.org/lib/fakes"
+	"code.cloudfoundry.org/lib/rules"
+	"code.cloudfoundry.org/silk/daemon"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+)
+
+var _ = Describe("Masquerader", func() {
+
+	var (
+		fakeIPTables     *lib_fakes.IPTablesAdapter
+		masquerader      *netrules.Masquerader
+		pluginController *lib.PluginController
+
+		containerIP string
+		daemonPort  string
+		leaseCIDR   string
+		vtepName    string
+	)
+
+	BeforeEach(func() {
+		containerIP = "10.255.11.11"
+		daemonPort = "8787"
+		leaseCIDR = "10.255.11.0/24"
+		vtepName = "meow-vtep"
+
+		fakeDelegator := &fakes.Delegator{}
+		fakeIPTables = &lib_fakes.IPTablesAdapter{}
+		pluginController = &lib.PluginController{
+			Delegator: fakeDelegator,
+			IPTables:  fakeIPTables,
+		}
+		masquerader = &netrules.Masquerader{
+			PluginController: pluginController,
+			VTEPName:         vtepName,
+			DaemonPort:       daemonPort,
+			ContainerIP:      containerIP,
+		}
+	})
+
+	Describe("AddIPMasq", func() {
+		Context("when customNoMasqueradeCIDRRange is NOT set", func() {
+			var fakeSilkDaemonServer *ghttp.Server
+			var silkDaemonResponseBytes []byte
+
+			BeforeEach(func() {
+				fakeSilkDaemonServer = ghttp.NewServer()
+				daemonPort := strings.Split(fakeSilkDaemonServer.Addr(), ":")[1]
+				masquerader.DaemonPort = daemonPort
+
+				silkDaemonResponse := daemon.NetworkInfo{
+					OverlaySubnet: leaseCIDR,
+				}
+
+				var err error
+				silkDaemonResponseBytes, err = json.Marshal(silkDaemonResponse)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				fakeSilkDaemonServer.Close()
+			})
+
+			Context("when the silk daemon responds with the lease", func() {
+				BeforeEach(func() {
+					fakeSilkDaemonServer.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", "/"),
+							ghttp.RespondWith(http.StatusOK, silkDaemonResponseBytes),
+						),
+					)
+				})
+
+				It("calls silk daemon to get the lease and uses that instead", func() {
```

Comment:
Do you want to add an additional test to validate the caching of m.lease, to ensure we only call silk daemon once per masquerader object?

Reply:
already did! https://github.com/cloudfoundry/silk-release/blob/6906e696d22d5ccc2c8788502f3fad4666f17ad1/src/code.cloudfoundry.org/cni-wrapper-plugin/netrules/masquerader_test.go#L224-L237

I only tested the IPAdd followed by IPDel direction because that is the way it is used in main.go. I can add more tests if you want though.